velox/exec/Task.cpp

```diff
@@ -996,17 +966,29 @@ void Task::driverClosedLocked() {
 }
 
 bool Task::checkIfFinishedLocked() {
-  if ((numFinishedDrivers_ == numTotalDrivers_) && isRunningLocked()) {
-    if (taskStats_.executionEndTimeMs == 0) {
-      // In case we haven't set executionEndTimeMs due to all splits depleted,
-      // we set it here.
-      // This can happen due to task error or task being cancelled.
-      taskStats_.executionEndTimeMs = getCurrentTimeMs();
+  if (isRunningLocked()) {
+    // TODO Add support for terminating processing early in grouped execution.
+    bool allFinished = numFinishedDrivers_ == numTotalDrivers_;
+    if (!allFinished && isUngroupedExecution() && !driverFactories_.empty()) {
```

Comment:
If driverFactories_ is empty(), then allFinished shall also be true? Thanks!

Reply:
Good point. It is actually not possible to have driverFactories_.empty() here. Will remove the check.